src/site/content/en/blog/javascript-and-google-search-io-2019/index.md

```diff
@@ -0,0 +1,169 @@
+       // this browser supports the Geolocation API, request location!
+       navigator.geolocation.getCurrentPosition(
+         function onSuccess(position) {
+           // we successfully got the location, show local content
+           loadLocalContent(position);
+         }, function onError() {
+           // we failed to get the location, show fallback content
+           loadGlobalContent();
+         });
+     } else {
+       // this browser does not support the Geolocation API, show fallback content
+       loadGlobalContent();
+     }
+   </script>
+ </body>
+
+```
+
+If you have problems with your JavaScript site, [walk through our troubleshooting guide](https://developers.google.com/search/docs/guides/fix-search-javascript) to find solutions.
+
+## Learn how Googlebot works
+
+Googlebot is a pipeline with different components. Let's take a look at this pipeline to understand how Googlebot indexes pages for Google Search.
+
+<figure class="w-figure w-figure--center w-figure--fullbleed">
+  <img src="googlebot-process.png" alt="A diagram showing a URL moving from a crawling queue to a processing step that extracts linked URLs and adds them to the crawling queue, a rendering queue that feeds into a renderer which produces HTML. The processor uses this HTML to extract linked URLs again and index the content.">
+  <figcaption class="w-figcaption w-figcaption--fullbleed">
+    Fig. 1 — Googlebot's pipeline for crawling, rendering and indexing a page.
+  </figcaption>
+</figure>
+
+The process works like this:
+
+1. Googlebot queues URLs for crawling.
+2. It then fetches the URLs with an HTTP request based on the [crawl budget](https://webmasters.googleblog.com/2017/01/what-crawl-budget-means-for-googlebot.html).
+3. Googlebot's next stage is scanning the HTML for links and queues the discovered links for crawling.
+4. Googlebot then queues the page for rendering.
+5. As soon as possible, a headless Chromium instance renders the page (JavaScript is executed here).
+6. Googlebot uses the rendered HTML to index the page.
+
+Your technical setup can influence the process of crawling, rendering, and indexing. For example, slow response times or server errors can impact the [crawl budget](https://webmasters.googleblog.com/2017/01/what-crawl-budget-means-for-googlebot.html) and links that are only rendered with JavaScript can lead to a slower discovery of these links.
+
+## Choose the right rendering strategy for your web app
+
+The default rendering strategy for single-page apps today is client-side rendering. The HTML loads the JavaScript which generates the content in the browser as it executes. 
+
+<figure class="w-figure">
+  <img src="spa-kittens.png" alt="A code box showing HTML that loads some scripts. A screenshot of a web page on mobile that shows placeholder images while loading the actual content.">
+  <figcaption class="w-figcaption">
+    The rendering strategy influences the performance and robustness of your web apps.
```

Comment:
I'd suggest putting something in the main body of the post that makes it clear what this image is showing--like you did in the talk, where you explained that this is an app for showing cat images, and all of it is generated by JS.